src/renderer/services/db/api/repositories/metrics.ts

```diff
@@ -14,5 +14,5 @@ export class MetricsRepository extends BulkRepository<Metrics> {
 
 
 export function generateMetricsKey(date: Date, method: string, instanceId: string): string{
-    return `${date.toISOString()}-${method}-${instanceId}`;
+    return `${instanceId}-${date.toISOString()}-${method}`;
```

Comment:
this date format doesn't really fit, we need something like `yyyymmddhh`

Reply:
Are you sure about this? It's probably better to stick with one of the date representation standards and ISO (YYYY-MM-DDTHH:mm:ss.sssZ) is pretty good for our case as it can be queried in multiple ways:
- filter by year (you can also set the year range)
```
gte: '<instanceId><selected-year>'
lte: '<instanceId><selected-year>~'
```

- filter by month (month range also available)
```
gte: '<instanceId><selected-year>-<selected-month>'
lte: '<instanceId><selected-year>-<selected-month>~'
```

- filter by minute
```
gte: '<instanceId><selected-year>-<selected-month>-<selected-day>T<selected-hour>:<selected-minute>'
lte: '<instanceId><selected-year>-<selected-month>-<selected-day>T<selected-hour>:<selected-minute>~'
```
- etc.

So ISO is basically very similar to what you are proposing but it is standardized and quite flexible.

Reply:
valid arguments, lrts keep it this way. my argument was that it makes filtering methods impossible but we have to filter mannually anyways

Reply:
Yeah, we can not query by the last parameter (methodName) in any case. 
We could possibly restrict the query for the method name by querying the events that happened in the last x days/months/years and then filter the result to extract entries that reference desired method.

Reply:
I think it's better to save it under different key but we don't need it for now (filtering by method)